### PR TITLE
[DSCP-475] Update sample config and set up `xelatex` for deployment.

### PR DESCRIPTION
**Pull request by flyingleafe**
### Description

Missed that when reviewed @Martoon-00 's pr on DSCP-475 :angry: 

Also includes a bunch of changes, mostly to `nix` files, which allow for deployment of `xelatex` with all necessary modules as well.

### YT issue

https://issues.serokell.io/issue/DSCP-475

### Checklist

Hint: a perfect PR has all the checkmarks set.

Possible related changes (conditional):
- [x] I added tests if required, in case they are:
  - Covering new introduced functionality (feature).
  - Regression tests preventing the bug from silently reappearing again (bugfix).
- [x] I have checked the [sample config](../tree/master/docs/config-full-sample.yaml) and [launch scripts](../tree/master/scripts/launch) and updated them if it was necessary (especially if executables or CLIs were changed).
- [x] I have checked READMEs and changed them accordingly if it's necessary (the main [README.md](../tree/master/README.md) as well as subproject READMEs for all related executables).
- [x] If a public API structure or/and data serialization was changed, I made sure that these changes are reflected in the:
  - [Swagger](../tree/master/specs/disciplina) API specs.
  - Any [related documentation](../tree/master/docs/api-types.md).
- [x] If any public documentation was written, I have spellchecked it.

Stylistic (obligatory):
- [x] My commit history is clean, descriptive and do not contain merge or revert commits or commits like "WIP".
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).
- [x] My code is documented (haddock) and these docs are decent (full enough and spellchecked).

### CODE REVIEW

File: docs/config-full-sample.yaml [hunk 1]
```diff
@@ -211,6 +211,10 @@ sample:
     publishing:
       period: 30s
 
+    # Parameters for certificates generator
+    certificates:
+      resources: "/var/lib/disciplina/certificate-templates"
```
Comment (Martoon-00) [inline docs/config-full-sample.yaml]: Although, that's a pity that now one can't simply run the node using this config.

File: educator/src/Dscp/Educator/Launcher/Resource.hs [hunk 2]
```diff
@@ -51,10 +53,22 @@ instance AllocResource (KeyResources EducatorNode) where
...
+            mPath <- liftIO $ findExecutable executable
+            Pdf.LatexPath <$>
+                nothingToThrow (ExecutableNotFound "LaTeX generator" executable)
+                mPath
+
 instance AllocResource Pdf.ResourcePath where
     type Deps Pdf.ResourcePath = (FilePath, AppDir)
     allocResource (userPath, appDir) =
-        buildComponentR "resources path" preparePath (\_ -> pass)
+        buildComponentR "LaTeX resources path" preparePath (\_ -> pass)
```
Comment (Martoon-00) [inline educator/src/Dscp/Educator/Launcher/Resource.hs]: Could you also remove `TODO` 10 rows below?

**Comment by Martoon-00** [comment]
Why have you decided to propagate the path to `xelatex` through the program? Nix cannot into `PATH` modification?

**Comment by flyingleafe** [comment]
@Martoon-00 it's either Nix or me + Nix who cannot into `PATH` modification for a _particular_ user `disciplina`, which runs the daemons on servers, uses `nologin` as a shell and therefore probably has no `PATH` at all :man_shrugging:

**Comment by Martoon-00** [comment]
Okay, life is hard :unamused:

**Pull request merged**